Jeremie/ri-902-les-liens-url-cassent-la-mise-en-page-quand-ils-sont-dans

ledjay opened the pull request:
# PR Description: Fix Grid Layout Issues in Indented Lists

## 🎯 Problem

The grid-based layout for indented lists was breaking because the `::before` pseudo-element (containing the list number/bullet) was at the same DOM level as each child element within the `<li>`. This caused grid column misalignment and layout issues.

### Root Cause:
```html
<li value="1">
  <!-- ::before pseudo-element renders here (grid-column: 1) -->
  <b>Bold text</b>      <!-- grid-column: 3 -->
  <span>Normal text</span>  <!-- grid-column: 3 -->
  <a>Link</a>           <!-- grid-column: 3 -->
</li>
```

With CSS like:
```html
ol li {
  display: grid;
  grid-template-columns: max-content 1ch 1fr;
}

ol li::before {
  grid-column: 1;  /* Number/bullet */
}

ol li > * {
  grid-column: 3;  /* Content */
}
```

Issue: Each child element (<b>, <span>, <a>) was treated as a separate grid item, causing them to stack vertically instead of flowing inline within the content column.

✨ Solution
1. Wrap List Item Content in Container (1cc6d27)
Added a transformation that wraps all content inside <li value="X"> elements in a single <div> container:

After:
```html
<li value="1">
  <!-- ::before pseudo-element (grid-column: 1) -->
  <div>                    <!-- grid-column: 3 - single container -->
    <b>Bold text</b>
    <span>Normal text</span>
    <a>Link</a>
  </div>
</li>
```

Result: Now the ::before marker and the content <div> are the only two grid items, allowing content to flow naturally inside the wrapper while maintaining proper grid alignment.

ledjay committed:
fix: wrap list items with value attribute in div for consistent styling

ledjay committed:
refactor: simplify Text.module.scss as styles already defined in prose-list.css

kaaloo commented on apps/client/src/components/Pages/dispositif/Text/Text.tsx:
```diff
@@ -29,11 +29,28 @@ const Text = (props: Props) => {
     setHasMounted(true);
   }, []);
 
+  // Wrap content of li elements with value attribute in a div
+  const transformListItems = (html: string): string => {
+    return html.replace(
+      /<li\s+value="(\d+)"([^>]*)>([\s\S]*?)<\/li>/gi,
+      (match: string, value: string, attrs: string, content: string) => {
+        const trimmedContent = content.trim();
+        // Skip if already wrapped in a div
+        if (trimmedContent.startsWith("<div>") && trimmedContent.endsWith("</div>")) {
+          return match;
+        }
+        return `<li value="${value}"${attrs}><div>${content}</div></li>`;
+      },
+    );
+  };
```
Perhaps double check this review comment @ledjay? Actually I think there may be a more robust way to update the html by using the DOMParse.parseFromString() method.  Gemini have me some sample code.  More robust but maybe overkill.

Convert HTML String to DOM

```javascript
const htmlString = '<ul><li>Item 1</li><li>Item 2</li><li>Item 3</li></ul>';
const parser = new DOMParser();
const doc = parser.parseFromString(htmlString, 'text/html');

// The main element you want to work with (e.g., the <ul>) is now within the <body>
const container = doc.body.firstChild; // This assumes the string is a snippet like above.
// For full HTML documents, you'd work with doc.documentElement (the <html> tag)
```

Surround &lt;li&gt; with &lt;div&gt;

```javascript
// 2. Find all <li> elements
const listItems = container.querySelectorAll('li');

listItems.forEach(li => {
    // 3. Create the new <div> element
    const divWrapper = doc.createElement('div');

    // 4. Move the <li> inside the new <div>
    divWrapper.appendChild(li.cloneNode(true)); // Clone to avoid unexpected side effects if the original list is still in use
    
    // 5. Replace the original <li> with the new <div> wrapper in the parent
    li.parentNode.replaceChild(divWrapper, li);
});
```

Convert DOM to string

```javascript
// Get the resulting HTML string from the modified container
const modifiedHtmlString = container.outerHTML;

console.log(modifiedHtmlString);
// Output: <ul><div><li>Item 1</li></div><div><li>Item 2</li></div><div><li>Item 3</li></div></ul>
```

ledjay replied:
Good point.
Will Check this out Thanks.

ledjay committed:
feat: improve list item handling and counter reset behavior

- Rewrote list item transformation logic to use DOMParser for more robust HTML manipulation
- Added check for existing div wrappers to prevent double-wrapping content
- Improved whitespace handling in list items by checking text node content
- Added !important to counter-reset CSS rules to ensure proper list numbering
- Memoized content conversion using useMemo to optimize performance
- Added server-side rendering safety check for DOMParser availability

ledjay commented:
@kaaloo fixed the dom rewrite with dom parser :)
